template.js

```diff
@@ -14,9 +14,102 @@ ${content}
     const anchor = document.createElement('a')
     anchor.setAttribute('href', '#' + element.id)
     anchor.setAttribute('class', 'agregore-header-anchor')
-    anchor.innerHTML = element.innerHTML
     element.innerHTML = anchor.outerHTML
   }
+
+  // Table sorting functionality
+  function addTableSorting() {
+    const tables = document.querySelectorAll('table')
+    
+    tables.forEach(table => {
+      const thead = table.querySelector('thead')
+      const tbody = table.querySelector('tbody')
+      
+      if (!thead || !tbody) return
+      
+      const headerCells = thead.querySelectorAll('th')
+      
+      headerCells.forEach((headerCell, columnIndex) => {
+        // Create sort button
+        const sortButton = document.createElement('button')
+        sortButton.innerHTML = '↕️'
+        sortButton.className = 'agregore-sort-btn'
+        sortButton.style.cssText = 'background: none; border: none; cursor: pointer; font-size: 1em; padding: 2px 6px; margin-left: 8px;'
+        sortButton.setAttribute('data-column', columnIndex)
+        sortButton.setAttribute('data-sort-state', 'unsorted')
+        
+        // Add click handler
+        sortButton.addEventListener('click', () => handleSort(table, columnIndex, sortButton))
+        
+        // Append button to header cell
+        headerCell.appendChild(sortButton)
+      })
+    })
+  }
+
+  function handleSort(table, columnIndex, clickedButton) {
+    const tbody = table.querySelector('tbody')
+    const rows = Array.from(tbody.querySelectorAll('tr'))
```

Comment:
Is `Array.from` necessary here? We should just use `for of` on the returned list and skip allocating this extra array